src/diff_tform.c

```diff
@@ -296,6 +296,9 @@ static int normalize_find_opts(
 	if (opts->flags & GIT_DIFF_BREAK_REWRITES)
 		opts->flags |= GIT_DIFF_FIND_REWRITES;
 
+	if (opts->flags & GIT_DIFF_FIND_NO_RENAMES)
+		opts->flags &= ~GIT_DIFF_FIND_ALL;
```

Comment:
Do you still want it to exit early? I just placed it where I thought it was appropriate.